app/src/main/java/com/ahmettekin/WeatherApp/adapter/RecyclerViewAdapter.java

```diff
@@ -99,12 +99,20 @@ private void initViews() {
 
         private void configureUI(WeatherItem weatherItem) {
 
-            itemView.setBackgroundResource(weatherItem.getWeather()[0].getSkyId());
+            WeatherItem.Weather weatherObject;
+
+            if (weatherItem.getWeathers().length > 0) {
+                weatherObject = weatherItem.getWeathers()[0];
+            } else {
+                weatherObject = null;
+            }
+
+            itemView.setBackgroundResource(weatherObject.getSkyId());
             tvWeatherDescription.setText(weatherItem.getName());
             tvWeatherTemp.setText(MessageFormat.format("Sıcaklık : {0}{1}", weatherItem.getMain().getTemp(), DEGREE_SYMBOL));
-            tvSky.setText(MessageFormat.format("Gökyüzü : {0}", UpperCaseWords(weatherItem.getWeather()[0].getDescription())));
+            tvSky.setText(MessageFormat.format("Gökyüzü : {0}", UpperCaseWords(weatherObject.getDescription())));
```

Comment:
Diğer düzeltmeleri gördüm eline sağlık 👍  fakat burası yine olmamış çünkü `weatherObject = null` olursa 110,115 ve burası crash olur o yüzden weatherObject'yi kullandığın yerleri if'in içinde yazsan daha iyi olur sanki :)
- 1.yöntem olarak bunu kullanabilirsin:
`if (weatherItem.getWeathers().length > 0) {
       weatherObject = weatherItem.getWeathers()[0];
       //weatherObject kullanılacak yerler buraya
`
- 2. yöntemde `!= null` kontrolü yapmak o da yorar bence gerek yok :)